parser/src/lib.rs

```diff
@@ -24,33 +25,67 @@ pub mod token;
 mod token_reader;
 
 pub async fn parse(text: &str) -> Result<Ast, Error> {
-    let (lexer_in, lexer_out): (Sender<Token>, Receiver<Token>) = mpsc::channel(32);
-    let (normalizer_in, normalizer_out): (Sender<Token>, Receiver<Token>) = mpsc::channel(32);
+    let span = trace_span!("parsing");
+    let _enter = span.enter();
+    debug!(text);
+    let channel = 32;
+    trace!("reading channel {channel:?}");
+    let (lexer_in, lexer_out): (Sender<Token>, Receiver<Token>) =
+        mpsc::channel(channel);
+    debug!("successfully connected to lexer");
+    let (normalizer_in, normalizer_out): (Sender<Token>, Receiver<Token>) =
+        mpsc::channel(channel);
+    debug!("successfully connected to normalizer");
 
     let context = MathContext::new();
+    trace!("created MathContext");
     let lexer = Lexer::new(lexer_in);
+    trace!("created Lexer");
     let mut normalizer = Normalizer::new(lexer_out, normalizer_in);
+    trace!("created Normalizer");
     let mut parser = Parser::new(normalizer_out, context);
+    trace!("created Parser");
```

Comment:
I'd maybe prefer if this tracing happened in the `new` functions, or not at all. Nothing can really go wrong here, we are just initializing structs.